### ISSUE TITLE
OpenSSL 3.1.0 fails to build for aarch64 with gcc 4.9.4

### ISSUE THREAD

**Issue by hgy59** [openssl/openssl#20963]
Since 3.1.0 the build for aarch64 with gcc 4.9.4 fails to build two asm files

```
crypto/modes/aes-gcm-armv8-unroll8_64.S: Assembler messages:
crypto/modes/aes-gcm-armv8-unroll8_64.S:4: Error: unknown architecture `armv8.2-a+crypto'
...
```

and 
```
crypto/sm3/sm3-armv8.S: Assembler messages:
crypto/sm3/sm3-armv8.S:13: Error: unknown architecture `armv8.2-a'
```

In OpenSSL 3.0.8 those files do not yet exist and the build succeeds.

I can successfully build OpenSSL 3.1.0 when patching the two files and replacing (downgrading) the archs to `armv8-a` and `armv8-a+crypto` (i.e. from arm8.2 to arm8).

Is it possible to fix OpenSSL 3.x to support older gcc for aarch64?

The commit https://github.com/SynoCommunity/spksrc/pull/5730/files#diff-b9489cdbf8e9bbfaed8bd51281ad3eb9e320f5e7e1e06f99ff2beeaef033b5f7 includes the patches of the temp. fix - but IMHO this should be fixed within openssl project by evaluation of the capabilities of the available compiler.

_as in a few months the EOL of OpenSSL 1.1.1 arrives, this should be fixed even for older compilers_

**Comment by hgy59** [comment]
The file `crypto/modes/aes-gcm-armv8-unroll8_64.S` was already mentioned in  #17795 and #17773 and was originally introduced in #15916.

**Comment by hgy59** [comment]
It was already planned to set `armv8.2-a+crypto` back to `armv8-a+crypto` in `aes-gcm-armv8-unroll8_64.S` in #15916 - but this was never done.

> 
>               It's manually encoded, not very necessary, will remove it in next commit.
> 
> _Originally posted by @xkqian in https://github.com/openssl/openssl/pull/15916#discussion_r788327782_

**Comment by zorrorffm** [comment]
Removing architecture declaration is doable,  we will take a look and fix it.

**Comment by hgy59** [comment]
> Sorry, this comment is for the code in file crypto/arm64cpuid.pl, if for the file aes-gcm-armv8-unroll8_64.S, they are not manuualy encoded. Anyway, we can provide one PR to fix the issue on gcc 4.9.4.

I am sorry, you are right, it was not for `aes-gcm-armv8-unroll8_6`. Anyway if this can be fixed for aes-gcm-armv8-unroll8_64.S and sm3-armv8.S I much appreciate it.
Otherwise I would need to disable asm for my situation and I want to avoid it.

**Comment by zorrorffm** [comment]
For sm3 assembly code, it's fine to remove the declaration of architecture feature `.arch armv8.2-a+crypto` since all the sm3 instructions that require the support from new compiler have been translated to instruction binaries, there is no compiler dependency any more.  For aes-gcm, the instructions under sha3 feature requires armv8.2-a support, but those instructions also have been translated into instruction binaries, so it's fine to downgrade `armv8.2-a` to `armv8-a`.  This change looks good to me.
